pkg/controller/account/credentials_rotator.go

```diff
@@ -71,6 +72,10 @@ func (r *ReconcileAccount) RotateCredentials(reqLogger logr.Logger, awsSetupClie
 
 	STSCredentialsSecret := CreateSecret(secretName, STSCredentialsSecretNamespace, secretData)
 
+	if err := controllerutil.SetControllerReference(account, STSCredentialsSecret, r.scheme); err != nil {
```

Comment:
Curious, is there a reason not to put this in `CreateSecret` instead?

Reply:
IMO, this is part of the creation of the secret

Reply:
Right.  And both at creation time and secret refresh time the secrets are looked for, deleted if found, and then created anew.  We could push all of that logic into the CreateSecret function as well.  I just have a separate card to do the refactor, so that we can get this fix in asap, and then refactor the functionality separately in case there's other gotchas that I haven't seen yet.

Reply:
Fair enough :+1: